src/operators/Remap/avtRemapFilter.C

```diff
@@ -131,447 +138,476 @@ avtRemapFilter::SetAtts(const AttributeGroup *a)
-        avtRemapRectilinearDensity->SetComponent(i, 0, newCellDensity);
-*/
-        
-        // --- Clean up --- //
-        subCellVolumes->Delete();
-        ug->Delete();
-        
-        for (int cdx = 0; cdx < 4; cdx++) { // future will be 6 for 3D
-            planeArray[cdx]->Delete();
-            funcsArray[cdx]->Delete();
-            clipperArray[cdx]->Delete();
-        } // end clipper loop
 
-    } // end rectilinear grid loop
-    
-std::cout << "DEBUG: max diff of cell volumes: " << DEBUG_maxDiff << std::endl;
-    
-        
-std::cout << "Populated vars in rg:" << std::endl;
-PrintData(rg);
-    
-    
-    // ----------------------------------------------------- //
-    // --- Attach the datasets onto the rectilinear grid --- //
-    // ----------------------------------------------------- //
-    
-    //rg->GetCellData()->AddArray(avtRemapRectilinearVolume);
-    //rg->GetCellData()->CopyFieldOn("avtRemapRectilinearVolume");
-    
-/*    avtRemapRectilinearDensity->SetName("d");
-    rg->GetCellData()->AddArray(avtRemapRectilinearDensity);
-    rg->GetCellData()->SetScalars(avtRemapRectilinearDensity);
-    rg->GetCellData()->CopyFieldOn("d");
-    avtRemapRectilinearDensity->Delete();
-*/
-    
-/*    for (int vdx = 0; vdx < nVariables; vdx++)
-    {
-        rg->GetCellData()->AddArray(vars[vdx]);
-        rg->GetCellData()->SetScalars(vars[vdx]);
-    }
-*/    
-    avtDataRepresentation *out_dr =
-            new avtDataRepresentation(rg, 0, "", false);
-            
-std::cout << "Created out_dr:" << std::endl;
-PrintData(out_dr);
-
-std::cout << "Types and numbers of cells in sub meshes" << std::endl;
-for (std::map<std::string,int>::const_iterator iter = DEBUG_CellTypeList.begin();
-        iter != DEBUG_CellTypeList.end(); ++iter)
+// ****************************************************************************
+//  Method: avtRemapFilter::MakeClippingFunction
+//
+//  Purpose:
+//      Generate vtkImplicitBooleans to store in the x, y, z array of functions
+//      that will be used for clipping.
+//
+//  Arguments:
+//      rCell      The id of the current cell.
+//      side       The "side" of the cell as an integer.
+//
+//  Programmer: rusu1
+//  Creation:   Wed Apr  3 13:52:34 PDT 2019
+//
+// ****************************************************************************
+
+void
+avtRemapFilter::MakeClippingFunction(int rCell, int side)
 {
-    if (iter->second > 0)
-    {
-        std::cout << iter->first << ": " << iter->second << std::endl;
+    debug4 << "avtRemapFilter::MakeFunction" << std::endl;
+    double cellBounds[6] = {0., 0., 0., 0., 0., 0.};
+    double origin[3] = {0., 0., 0.};
+    double normal[3] = {0., 0., 0.};
+    rg->GetCell(rCell)->GetBounds(cellBounds);
+    origin[side/2] = cellBounds[side];
+    normal[side/2] = 1.;
+
+    vtkPlane* plane = vtkPlane::New();
+    plane->SetOrigin(origin);
+    plane->SetNormal(normal);
+    vtkImplicitBoolean* funcs = vtkImplicitBoolean::New();
```

Comment:
Is the memory for funcs ever cleaned up?(deleted)?

Reply:
Fixed